crates/delegator/src/main.rs

```diff
@@ -51,71 +40,36 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
-    let mut stdin = BufReader::new(stdin()).lines();
+    Delegator::new(job_witness_tx, swarm_events_rx);
 
-    loop {
-        tokio::select! {
-            Ok(Some(_)) = stdin.next_line() => {
-                // TODO: handle fixture better way
-                let fixture = fixture();
-                compiler_scheduler.push(compiler.run(fixture.program_path.clone(), fixture.program_input_path)?);
-                info!("Scheduled compiling program at path: {:?}", fixture.program_path);
+    let server = DelegatorGRPCServer::new(
+        node_account.get_signing_key().to_owned(),
+        new_job_topic_tx,
+        job_witness_rx,
+    );
 
-            },
-            Some(event) = message_stream.next() => {
-                match event {
-                    Event::Message { message, .. } => {
-                        // Received a new-job message from the network
-                        if message.topic ==  gossipsub_ident_topic(Network::Sepolia, Topic::NewJob).into() {
-                            let job: Job = serde_json::from_slice(&message.data).unwrap();
-                            info!("Received a new job event: {}", hash!(&job));
+    let (mut health_reporter, health_service) = tonic_health::server::health_reporter();
+    health_reporter.set_serving::<DelegatorServiceServer<DelegatorGRPCServer>>().await;
 
-                        }
-                        // Received a picked-job message from the network
-                        if message.topic ==  gossipsub_ident_topic(Network::Sepolia, Topic::PickedJob).into() {
-                            let job: Job = serde_json::from_slice(&message.data).unwrap();
-                            info!("Received picked job event: {}", hash!(&job));
-                        }
-                    },
-                    Event::Subscribed { peer_id, topic } => {
-                        info!("{} subscribed to the topic {}", peer_id.to_string(), topic.to_string());
-                    },
-                    Event::Unsubscribed { peer_id, topic }=> {
-                        info!("{} unsubscribed to the topic {}", peer_id.to_string(), topic.to_string());
-                    },
-                    _ => {}
-                }
-            },
-            Some(Ok(event_vec)) = event_stream.next() => {
-                debug!("{:?}", event_vec);
-            },
-            Some(Ok(job)) = compiler_scheduler.next() => {
-                let serialized_job = serde_json::to_string(&job).unwrap();
-                send_topic_tx.send(serialized_job.into()).await?;
-                info!("Sent a new job: {}", hash!(&job));
-            },
-            _ = shutdown_signal() => {
-                break
-            }
-            else => break
-        }
-    }
+    Server::builder()
+        .add_service(health_service)
+        .add_service(DelegatorServiceServer::new(server))
+        .serve_with_shutdown("0.0.0.0:50051".parse().unwrap(), shutdown_signal())
```

Comment:
for supporting ip4 / ip6 both,
```suggestion
        .serve_with_shutdown("[::]:50051".parse().unwrap(), shutdown_signal())
```